theory/theory.md

```diff
@@ -0,0 +1,362 @@
+
+Inheritance is a useful feature because inheritance can also help with code reuse without having to modify them. In some circumstances, the inheritance structure reflects the problem's natural nature, making the programme easier to understand.
+
+Inheritance, on the other hand, can make programmes harder to read. When a method is called, it is not always obvious where to look for its definition. The relevant code may be spread over numerous modules. Furthermore, many of the things that may be done using inheritance may be done just as neatly without inheritance. 
+
+Following is an example of inheritance where a class named Car inherited properties and methods from a class named Vehicle.
+
+Create a class named Vehicle, with brand and name properties, and a printname method:
+
+    class Vehicle:
+      def __init__(self, brand, name):
+        self.brand = brand
+        self.name = name
+    
+      def printname(self):
+        print(self.brand, self.name)
+    
+    x = Vehicle("Toyota", "Sora Bus")
+    x.printname()
+
+    # Create a class named Car, which will inherit the properties and methods from the Vehicle class:
+    class Car(Vehicle):
+      pass
+    
+    # Now the Car class has the same properties and methods as the Vehicle class.
+    x = Car("Ford", "F-Series")
+    x.printname()
+
+Source : https://openbookproject.net/thinkcs/python/english3e/inheritance.html#:~:text=Inheritance%20is%20the%20ability%20to,methods%20of%20the%20existing%20class.
+
+## 4. If you had to make a program that could vote for the top three funniest people in the office, how would you do that? How would you make it possible to vote on those people?
+
+Given the limited information provide, I will make the following assumptions
+- Funny people are already nominated
+- People selects one person from the nominated people
+- The person who has the most votes are considered to be the funniest
+- We don’t need to consider a tie
+- Vote eligibility check is not required. 
+
+I would make it possible to vote on those people by creating a website, which presents the options, maybe buttons with nominated people's name. When a voter clicks the button, the website will send the vote and the data is stored in an appropriate storage, database or a file depending on the number of voters.
+
+For the backend, I would use a lightweight framework such as Flask if the number of voters are small. Otherwise, I would consider other frameworks such as Django.  
+
+If it is stored in a file, I would use an array to store the vallots and use Counter from collections to count the votes. The results will be sorted and the top three people with highest votes are the top three funniest people.  
+
+If it is stored in a database such as sql, I would run a query COUNT(vote) and return top 3 people for the results. 
+
+Since there is no requirement to display the result, I would probably just print the results after the voting has completed. 
+

```

Comment:
Q4. 10 / 10
A wonderful high-level approach to developing this software. More information, could be provided but for now, this is enough to give one a grasp of the direction you would follow.